fix: Fixes for build-tools for LTS branch

tylerbutler opened the pull request:
The LTS branch is in quite a different state than the main branch, and as such the build-tools have regressed such that they don't work on that branch. I made the following changes to accommodate:

1. Fixed a bug that caused the `--root` flag to be ignored.
2. If the external config file is not found, use the git root as the inferred root.

## Remaining issues

This PR does not address the fact that `flub generate typetests` was removed after version 0.10.0. The command intended to replace doesn't work on LTS right now because it can't find the previous package. This needs to be debugged further and fixed but that can be done separately.

tylerbutler committed:
fix: Fixes for build-tools for LTS branch

curtisman committed:
Use `fluid-build` for per package build script to correct dependencies (#15903)

tylerbutler committed:
revert some cdup changes

tylerbutler committed:
tweaks

tylerbutler committed:
fix

tylerbutler committed:
fix

tylerbutler commented on build-tools/packages/build-tools/src/common/fluidUtils.ts:
```diff
@@ -55,9 +56,18 @@ async function isFluidRoot(dir: string) {
+		const gitRoot = childProcess
+			.execSync("git rev-parse --show-toplevel", { encoding: "utf8" })
+			.trim();
+		fluidConfig = path.join(gitRoot, "/package.json");
```
```suggestion
		fluidConfig = path.join(gitRoot, "package.json");
```

tylerbutler committed:
Update build-tools/packages/build-tools/src/common/fluidUtils.ts

markfields committed:
Delete tests which are both redundant and skipped (#15900)

ssimic2 reviewed (approved):
I am assuming you are moving the rest of the LTS fixes through separate PRs?

curtisman committed:
Non-zero exit when there is build error (#15919)

tianzhu007 committed:
Re-organize Throttling Order and Check maxNumberOfClientsPerDocument  (#15921)

Move throttle before verifyStorageToken.
Check the maxNumberOfClientsPerDocumenth right after throttling check
for connect doc

tylerbutler committed:
Merge branch 'main' into bt-lts-fixes

tylerbutler committed:
Merge branch 'main' into bt-lts-fixes

tylerbutler commented:
> I am assuming you are moving the rest of the LTS fixes through separate PRs?

Yeah, I am working through those changes in #15909.
